container.go

```diff
@@ -220,80 +236,128 @@ func (c Container) Call(function interface{}) error {
 }
 
 // Resolve takes an abstraction (reference of an interface type) and fills it with the related concrete.
-func (c Container) Resolve(abstraction interface{}) error {
-	return c.NamedResolve(abstraction, "")
+func (c *Container) Resolve(abstraction interface{}) error {
+	return c.ResolvedNamed(abstraction, "")
 }
 
-// NamedResolve takes abstraction and its name and fills it with the related concrete.
-func (c Container) NamedResolve(abstraction interface{}, name string) error {
-	receiverType := reflect.TypeOf(abstraction)
-	if receiverType == nil {
+// ResolveWithContext takes an abstraction and a context and fills it with the related concrete.
+func (c *Container) ResolveWithContext(ctx context.Context, abstraction interface{}) error {
+	err := c.RegisterSingleton(func() context.Context {
```

Comment:
We should not do this, instead we should teach `make` to take a context and have it understand that when the first argument of a resolver is of type `context.Context` that it calls the resolver using the context passed in to `make`.

Reply:
Updated implementation.